### ISSUE TITLE
Rework druid documentation

### ISSUE THREAD

**Issue by shreyas-s-rao** [gardener/etcd-druid#592]
**Feature (What you would like to be added):**
Rework druid documentation.

**Motivation (Why is this needed?):**
The current documentation for etcd-druid is not optimal for understanding the internal workings of druid, or for new contributors to start contributing to the project. Furthermore, there is a need to extensively document the testing process to allow for easier local testing for developers.

This may be a stretch, but we can also consider adding a new `DEP` concept, ie, `Druid Enhancement Proposal`, similar to [Gardener GEPs](https://github.com/gardener/gardener/tree/master/docs/proposals). This can allow contributors to first raise an enhancement proposal for new features which are large enough, and have extensive discussions on the feature before accepting the feature proposal and starting work on the implementation tasks.

Additional tasks:
- [ ] update information about etcd-wrapper

**Comment by ishan16696** [comment]
I've created 2 issues related to this, I will create more and add them here.

- [ ] https://github.com/gardener/etcd-druid/issues/694
- [ ] https://github.com/gardener/etcd-druid/issues/695

**Comment by renormalize** [comment]
@shreyas-s-rao @ishan16696 #694 and #695 are both now closed.
What is the current opinion on `DEP` and updating the information about `etcd-wrapper`?

**Comment by shreyas-s-rao** [comment]
@renormalize DEP concept has been introduced via https://github.com/gardener/etcd-druid/blob/master/docs/proposals/04-etcd-member-custom-resource.md. The older proposals still need to be adapted to the DEP format.

Information about etcd-wrapper still needs to be updated in druid docs - specifically under `operations`, since knowledge about etcd-wrapper is required in order to interact with the etcd container deployed via druid.

**Comment by unmarshall** [comment]
@shreyas-s-rao do we need this issue now? We have already revamped etcd-druid documentation as part of https://github.com/gardener/etcd-druid/pull/909

**Comment by shreyas-s-rao** [comment]
You're right, we now mark this is completed, as all documentation enhancements have been taken care of in issue #813 and PR #909 .

/close
